File: 0053-maximum-subarray/0053-maximum-subarray.c
#define MAX(a, b) a > b ? a : b

int maxSubArray(int* nums, int numsSize){
    int sum = 0;
    int best = INT_MIN;
    
    for (int i = 0; i < numsSize; i++) {
        sum = MAX(sum + nums[i], nums[i]);
        best = MAX(best, sum);
    }
    return best;
}

File: 0053-maximum-subarray/0053-maximum-subarray.go
func max(a int, b int) int {
    if a > b {
        return a
    }
    return b
}

func maxSubArray(nums []int) int {
    best := nums[0]
    sum := 0
    
    for _, val := range nums {
        sum = max(val, sum + val)
        best = max(best, sum)
    }
    return best
}